src/commands/registerCommands.ts

```diff
@@ -0,0 +1,104 @@
+import { commands, ExtensionContext, window } from "vscode";
+import { VSCodeCommands } from "../definitions/constants";
+import { ProjectLabelInfo } from "../definitions/ProjectLabelInfo";
+import { requestStandardMode } from "../utils/requestStandardMode";
+import { sendCommandFailedTelemetry, sendCommandSucceededTelemetry } from "../utils/telemetryUtils";
+import { WelcomeWebview } from "../webviews/WelcomeWebview";
+import { addExtensionsWizard } from "../wizards/addExtensions/addExtensionsWizard";
+import { startDebugging } from "../wizards/debugging/startDebugging";
+import { generateProjectWizard } from "../wizards/generateProject/generationWizard";
+
+const NOT_A_QUARKUS_PROJECT = new Error('No Quarkus projects were detected in this folder');
+const STANDARD_MODE_REQUEST_FAILED = new Error('Error occurred while requesting standard mode from the Java language server');
+
+/**
+ * Register the vscode-quarkus commands
+ *
+ * @param context the extension context
+ */
+export function registerVSCodeCommands(context: ExtensionContext): void {
+
+  /**
+   * Command for creating a Quarkus project
+   */
+  registerCommandWithTelemetry(context, VSCodeCommands.CREATE_PROJECT, generateProjectWizard);
+
+  /**
+   * Command for adding Quarkus extensions to current Quarkus Maven project
+   */
+  registerCommandWithTelemetry(context, VSCodeCommands.ADD_EXTENSIONS, withStandardMode(addExtensionsWizard, "Adding extensions"));
+
+  /**
+   * Command for debugging current Quarkus Maven project
+   */
+  registerCommandWithTelemetry(context, VSCodeCommands.DEBUG_QUARKUS_PROJECT, withStandardMode(startDebugging, "Debugging the extension"));
+
+  /**
+   * Command for displaying welcome page
+   */
+  registerCommandWithTelemetry(context, VSCodeCommands.QUARKUS_WELCOME, async () => { WelcomeWebview.createOrShow(context); });
+
+}
+
+/**
+ * Register a command with the given name and async function
+ *
+ * Displays errors raised by the function as error messages,
+ * and sends telemetry when the command succeeds or fails.
+ *
+ * @param context the extension context
+ * @param commandName the name of the command to register
+ * @param commandAction the async function to run when the command is called
+ */
+async function registerCommandWithTelemetry(context: ExtensionContext, commandName: string, commandAction: () => Promise<any>): Promise<void> {
+  context.subscriptions.push(commands.registerCommand(commandName, async () => {
+    try {
+      await commandAction();
+      sendCommandSucceededTelemetry(commandName);
+    } catch (e) {
```

Comment:
is `e` ever a String in our code ? the arguments of the 2 methods below expect e to be a string but afaict it's most likely going to be an Error, so seems like the type should be tested, or `e.msg` should be used.

Reply:
I tested the type to get the correct error message. I don't know if any of the libraries we use throw strings instead of errors when there is an error.

Reply:
Yeah I noticed we throw some Strings in our code but mostly I could see Errors for sure. I was just throwing an Error right before calling a command to try and see if it got picked up in telemetry, but it was coming back empty. This should fix it.